sonic_y_cable/y_cable.py

```diff
@@ -0,0 +1,341 @@
+    raise ImportError(str(e) + " - required module not found")
+
+# Global logger instance for helper functions and classes to log
+helper_logger = logger.Logger(SYSLOG_IDENTIFIER)
+
+# Global platform_chassis instance to call get_sfp required for read/write eeprom
+platform_chassis = None
+
+# definitions of the offset with width accomodated for values
+# of MUX register specs of upper page 0x04 starting at 640
+# info eeprom for Y Cable
+Y_CABLE_IDENTFIER_LOWER_PAGE = 0
+Y_CABLE_IDENTFIER_UPPER_PAGE = 128
+Y_CABLE_DETERMINE_CABLE_READ_SIDE = 640
+Y_CABLE_CHECK_LINK_ACTIVE = 641
+Y_CABLE_SWITCH_MUX_DIRECTION = 642
+Y_CABLE_ACTIVE_TOR_INDICATOR = 645
+Y_CABLE_MANUAL_SWITCH_COUNT = 669
+
+
+def load_platform_chassis_for_y_cable():
+    global platform_chassis
+
+    """
+    This call is neccessary to load and store the global platform_chassis
+    variable which provides the utiliy to call the Sfp_Base's class
+    apis of reading and writing eeprom which are required for API's
+    of the Y cable.
+    """
+
+    try:
+        platform_chassis = sonic_platform.platform.Platform().get_chassis()
+        helper_logger.log_info("chassis loaded {}".format(platform_chassis))
+    except Exception as e:
+        helper_logger.log_warning("Failed to load chassis due to {}".format(repr(e)))
+
+
+def toggle_mux_to_torA(physical_port):
+    """
+    This API specifically does a hard switch toggle of the Y cable's MUX regardless of link state to
+    TOR A. API basically writes to upper page 4 offset 130 the value of 0x2 and expects the MUX to toggle
+    to TOR A. Bit 0 value 0 means TOR A
+
+    Register Specification at offset 130 is documented below
+
+    Byte offset   bits     Name                    Description
+    130           7-2      Reserved                Reserved
+                  1        Hard vs. soft switch    0b0 - Switch only if a valid TOR link on target; 0b1 Switch to new target regardless of link status
+                  0        Switch Target           Switch Target - 0b0 - TOR#1, 0b1 - TOR#2; default is TOR #1
+
+    Args:
+         physical_port:
+             an Integer, the actual physical port connected to Y end of a Y cable which can toggle the MUX
+
+    Returns:
+        a Boolean, true if the toggle succeeded and false if it did not succeed.
+    """
+
+    buffer = bytearray([2])
+    curr_offset = Y_CABLE_SWITCH_MUX_DIRECTION
+
+    if platform_chassis is not None:
+        result = platform_chassis.get_sfp(
+            physical_port).write_eeprom(curr_offset, 1, buffer)
+    else:
+        helper_logger.log_error("platform_chassis is not loaded, failed to toggle mux to TOR A")
+        return False
+
+    return result
+
+
+def toggle_mux_to_torB(physical_port):
+    """
+    This API specifically does a hard switch toggle of the Y cable's MUX regardless of link state to
+    TOR B. API basically writes to upper page 4 offset 130 the value of 0x3 and expects the MUX to toggle
+    to TOR B. Bit 0 value 1 means TOR B
```

Comment:
Please also describe what it means to "toggle the MUX to TOR B", i.e., exactly what does this do, and how it affects other functions.

Reply:
fixed